nx/lib/nx/defn.ex

```diff
@@ -722,7 +791,13 @@ defmodule Nx.Defn do
         %{}
       end
 
-    exports = Map.put(exports, {name, arity}, %{kind: kind, defaults: defaults})
+    exports =
+      Map.put(exports, {name, arity}, %{
+        function_type: function_type,
```

Comment:
```suggestion
        type: function_type,
```

We know it is a function. :)